### PR TITLE
fix(await-async-utils): false positive when destructuring

### LINKED ISSUE #715: False positive for await-async-utils
### Have you read the Troubleshooting section?

Yes

### Plugin version

v5.9.1

### ESLint version

v7.32.0

### Node.js version

v14.18.1

### package manager and version

6.14.17

### Operating system

WSL2

### Bug description

On creating a wrapper around a testing-library async util, eslint triggers the false positive rule `await-async-utils`.

### Steps to reproduce

I created the following helper in my setup function:

```typescript
function setup() {
  mockedGetCurrentUserProfile.mockResolvedValue(INITIAL_VALUES);
  const utils = render(<ProfileSettingsPanel open onClose={jest.fn()} />);

  const waitForLoadComplete = () => {
    return waitForElementToBeRemoved(screen.queryByRole('progressbar'));
  };

  return { waitForLoadComplete, ...utils };
}
```

When I try to use it inside a test:
```typescript
it('example', () => {
    const { user, waitForLoadComplete } = setup();
               // ^^^^^^^^^^^^^ -> Promise returned from waitForLoadComplete wrapper over async util must be handled
    await waitForLoadComplete();
// ...
});
```

### Error output/screenshots

_No response_

### ESLint configuration

```javascript

const path = require('path');
const { printSchema } = require('graphql');
const { loadSchemaSync } = require('@graphql-tools/load');
const { GraphQLFileLoader } = require('@graphql-tools/graphql-file-loader');
const hq = require('alias-hq');

const webpackMap = hq.get('webpack');
const aliasMap = Object.keys(webpackMap).map((k) => [k, webpackMap[k]]);

const schemaPath = path.join(__dirname, './client/apollo/schema.graphql');
const schema = loadSchemaSync(schemaPath, {
  loaders: [new GraphQLFileLoader()],
});
const schemaString = printSchema(schema);

/** @type {eslint/ESLint.ConfigData} */
const config = {
  ignorePatterns: [
    'client/apollo/autogenTypes.ts',
    'client/apollo/schema.graphql',
    'server/index.ts',
  ],
  extends: ['@private-company', 'plugin:@next/next/recommended'],
  plugins: ['graphql', 'import'],
  settings: {
    /**
     * Path aliasing needs specific configuration.
     * This depends on
     * https://www.npmjs.com/package/eslint-import-resolver-alias
     */
    'import/resolver': {
      alias: {
        map: aliasMap,
        extensions: ['.ts', '.tsx', '.js', '.jsx', '.json'],
      },
    },
  },
  rules: {
    // From eslint-plugin-graphql
    'react/jsx-no-target-blank': 0,
    'no-console': 'error',
    '@typescript-eslint/no-unused-vars': 'error',
    'graphql/template-strings': [
      'error',
      {
        env: 'apollo',
        schemaString,
      },
      {
        env: 'literal',
        schemaString,
      },
    ],
  },
  overrides: [
    {
      files: ['*.ts', '*.tsx'],
      rules: {
        '@typescript-eslint/explicit-module-boundary-types': 'warn',
      },
    },
    {
      files: ['.eslintrc.js', 'jest.config.js'],
      rules: {
        'import/no-extraneous-dependencies': 'off',
      },
    },
  ],
};

module.exports = config;
```

### Rule(s) affected

await-async-utils

### Anything else?

_No response_

### Do you want to submit a pull request to fix this bug?

No

### PR DESCRIPTION
**Pull request by patriscus**
## Checks

- [x] I have read the [contributing guidelines](https://github.com/testing-library/eslint-plugin-testing-library/blob/main/CONTRIBUTING.md).

## Changes

- Use early return via a condition `isAsyncUtilOrKnownAliasAroundIt` to reduce indentation
- Upon visiting `VariableDeclarator` nodes, detect destructured property aliases and assignments of known async wrapper functions
- Add `getMessageId` function
- Add test cases

## Context

Fixes #715

### CODE REVIEW

File: tests/lib/rules/await-async-utils.test.ts [hunk 1]
```diff
@@ -258,6 +258,40 @@ ruleTester.run(RULE_NAME, rule, {
         test('edge case for no innermost function scope', () => {
           const foo = waitFor
         })
+      `,
+		},
+		{
+			code: `
+        function setup() {
```
Comment (patriscus) [inline tests/lib/rules/await-async-utils.test.ts:265]: Please thoroughly review the test cases any let me know if some are incorrectly marked as valid and vice versa :) 

Also:  Should I make use of the `...ASYNC_UTILS.map()` pattern used in other test cases in my new test cases as well?
*

**Comment by patriscus** [comment]
I'd also appreciate it if someone could also help me in figuring out why the lint step fails - seems like it's around files I haven't even touched? 🤔

**Comment by Belco90** [comment]
Thanks @patriscus! I'll try to take a look in the following days.

The linting errors reported... I'm afraid that's because we don't pin the dev dependencies, so a new version got installed, reporting more errors. I'll fix that in a separate PR, sorry about that.

**Comment by Belco90** [comment]
Linting errors should be gone now if you update your branch with `main` one.

File: tests/lib/rules/await-async-utils.test.ts (hunk 1 above)
Comment (Belco90) [reply tests/lib/rules/await-async-utils.test.ts:265]: Yes, you should wrap your test cases in `...ASYNC_UTILS.map()` so all async utils are automatically tested.

**Review by Belco90 — DISMISSED** [review]
Whoa, the PR looks great! You just need to iterate the tests over the `...ASYNC_UTILS.map()` and should be ready to go. Thanks!

**Review by Belco90 — COMMENTED** [review]
Approved by mistake.

File: tests/lib/rules/await-async-utils.test.ts (hunk 1 above)
Comment (patriscus) [reply tests/lib/rules/await-async-utils.test.ts:265]: @Belco90, I've changed the test cases in the latest commit by
1. Using `...ASYNC_UTILS.map()`
2. Renaming `waitForElementToBeRemoved` to `waitForAsyncUtil` inside the test cases (just a small change in order to align it better)

Feel free to have another look :)

**Review by Belco90 — APPROVED** [review]
Amazing! Looking great now, thanks for your contribution!
